dso-l2/src/main/java/com/tc/objectserver/handler/ProcessTransactionHandler.java

```diff
@@ -106,6 +108,12 @@ public synchronized void reconnectComplete() {
     public void handleEvent(TCMessage context) throws EventHandlerException {
       NodeID destinationID = context.getDestinationNodeID();
       invokeReturn.remove((ClientID)destinationID, context);
+      if(context instanceof VoltronEntityMultiResponse) {
```

Comment:
That is only for the failed path, though.  Additionally, what about non-invoke responses?  Do they all go through the multi-response path?  (while I would assume that they should, I seem to remember there being a reason why they didn't).

Additionally, can that call be moved from line 323 to here?  As it stands, we have a hidden assumption that these 2 paths are equivalent, even though they are in very different parts of the code, are run in different threads, and are not documented as such.